src/components/DeckMechanisms/canCardBePlayed.js

```diff
@@ -1,26 +1,31 @@
+import areCardsEqual from '../../helpers/areCardsEqual'
+
 /**
  * Return whether a given card can be played in the current state.
  * @param {Object} state - State being mutated
- * @param {String} id - Id of the card
+ * @param {Object} card - Object containing id and idx of the card
  * @return {Boolean} Whether the card can be played
  */
-const canCardBePlayed = (state, id) => {
-  const card = state.deck.find(card => card.id === id)
-  const isAffordable = card.mana <= state.mana
+const canCardBePlayed = (state, card) => {
+  if (!card) {
```

Comment:
If `card` ends up being an empty object (`{}`), `!{}` will not yield `true`, because `{}` is not considered falsey by default in JS. You might want to effectively check the `card.id` here.